### PR TITLE
Feature/update msal framework

### PR DESCRIPTION
**Pull request by jaelim-ms**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [X] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Update MSAL.framework to switch dynamic framework to static framework (library)

### CODE REVIEW

File: Vendor/macOS/MSAL/MSAL.framework/Versions/A/Headers/MSALPublicClientApplication.h [hunk 1]
```diff
@@ -225,7 +225,8 @@
  */
 - (void)allAccountsFilteredByAuthority:(nonnull MSALAccountsCompletionBlock)completionBlock;
 
-#pragma SafariViewController Support
+#pragma mark -
```
Comment (dhei) [inline Vendor/macOS/MSAL/MSAL.framework/Versions/A/Headers/MSALPublicClientApplication.h]: nit: Is `pragma mark -` needed?

Comment (jaelim-ms) [reply Vendor/macOS/MSAL/MSAL.framework/Versions/A/Headers/MSALPublicClientApplication.h]: That's the fix for the issue that we have. 😄

File: Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh [hunk 2]
```diff
@@ -2,4 +2,16 @@
 
 source $(dirname "$0")/../build-ios-framework.sh AppCenterIdentity
 
+# Add MSAL.
```
Comment (jwallra) [inline Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh]: please tab appropriately (2 spaces for consistency in Apple SDK)

File: Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh [hunk 3]
```diff
@@ -2,4 +2,16 @@
...
+if [ -z $(otool -L "${INSTALL_DIR}/${FMK_NAME}" | grep 'libMSAL') ]
+then
+if [ -z "$MS_ARM64E_XCODE_PATH" ] || [ ! -d "$MS_ARM64E_XCODE_PATH" ] ; then
```
Comment (jwallra) [inline Scripts/AppCenterIdentity/build-appcenteridentity-ios-framework.sh]: Please split lines (if / then) or combine lines 6-7 for consistency

File: AppCenterIdentity/AppCenterIdentity/Support/iOS.xcconfig [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #include "../../../Config/iOS.xcconfig"
 
-OTHER_LDFLAGS=$(inherited) -framework UIKit
+OTHER_LDFLAGS=$(inherited) -framework UIKit -framework AuthenticationServices
```
Comment (ElektrojungeMS) [inline AppCenterIdentity/AppCenterIdentity/Support/iOS.xcconfig]: I wanted to give the following feedback

> We need to add `AuthenticationServices` to the podspec of identity, too.

but then realized that we need to actually add App Center Identity to our podspec, which – I think – should be its own PR.

Comment (jaelim-ms) [reply AppCenterIdentity/AppCenterIdentity/Support/iOS.xcconfig]: Yeah.. We don't have the podspec for Identity now, it will be added at the time we add Identity to podspec. I'll capture this as a bug.
